### PR TITLE
fix(queue): avoid panic when debug-logging InsertManyFast results

### LINKED ISSUE #12225: [Bug]: Actions V2 Regression: 'Event' condition executions cause widespread API errors and break Login
### Preflight Checklist

- [x] I could not find a solution in the documentation, the existing issues or discussions
- [x] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

v4.11.0 - v4.15.0

### Database

PostgreSQL

### Database Version

17

### Describe the problem caused by this bug

When configuring an Actions V2 Target and an Execution bound to an **Event** condition (especially triggering on all events), it causes widespread API failures across the ZITADEL instance. This happens indiscriminately, even breaking the main login page. 

If `Interrupt on Error` is enabled on the Target, the ZITADEL instance becomes completely locked and unusable because every API call results in an error.

**Important observations:**
* The ZITADEL API fails **regardless of the webhook's response**. 
* Whether the webhook correctly returns an `HTTP 200 OK` with a valid JSON body, returns an error, or if the Target URL is **completely unreachable**, the outcome from ZITADEL is the exact same API error surfacing on the frontend/client.
* The external webhook *is* being triggered. Webhook logs show it receives the event payload and returns `200 OK`, and ZITADEL's Event Logs show the events being processed, but the original ZITADEL API call that triggered the action always fails.
* **This is a regression:** This feature worked perfectly fine in `v4.10.1` when Actions V2 was in beta. The issue is consistently present in later versions including new `v4.15.0` .

### To reproduce

1. Create a simple Express.js webhook (code below) that logs the request and returns a `200 OK` with JSON.
2. Go to **ZITADEL Console > Default Settings > Actions > Targets**.
3. Create a Target pointing to the webhook. I have tested all target types: **REST Call**, **REST Webhook**, and **REST Async**.
4. Go to **ZITADEL Console > Default Settings > Actions > Actions** (Executions). 
5. Create an Execution bound to the target, and set the condition type specifically to **Event**, configured to trigger on all events.
6. Attempt to use ZITADEL normally (e.g., loading the login page, creating a user, or making any API calls).
7. **Result:** The API calls immediately throw errors and fail. 

### Screenshots

Image

Image

### Expected behavior

When an **Event** webhook legitimately returns an `HTTP 200 OK` (or even if a REST Async target is unreachable), ZITADEL should process the event asynchronously/synchronously without indiscriminately failing the original API call. This worked as expected in earlier beta versions like `v4.10.1`. Enabling Actions V2 with an Event condition should not break the system's core API routes and lock administrators out.

### Operating System

_No response_

### Relevant Configuration

_No response_

### Code Snippet (Target Webhook)
Here is the minimal Node/Express app used to test the target webhook. It simply logs the payload and returns `200 OK` with JSON.

```javascript
const express = require("express");
const app = express();

app.use(express.json());

app.post("/webhook", (req, res) => {
  try {
    console.log("Webhook triggered!");
    console.log(JSON.stringify(req.body, null, 2));

    // Simply returning a success status
    res.status(200).json({});
  } catch (err) {
    console.error("Error processing webhook:", err);
    res.status(500).send("error");
  }
});

const PORT = 8088;
app.listen(PORT, () => {
  console.log(`Webhook server listening on http://localhost:${PORT}`);
});
```

### PR DESCRIPTION
**Pull request by RmdanJr**
# Which Problems Are Solved

- Actions V2 **Event** executions can fail the originating API with `Errors.Internal (RECOVER)` / a nil-pointer panic when `Log.Level` is `debug`, even though the domain operation and target side effects may already have succeeded (for example Create User / Create Session while a webhook still fires). Relates to #12225.
- Root cause: River `InsertManyFast` / `InsertManyFastTx` returns a count-sized slice of nil `*JobInsertResult` (no job rows). The queue log middleware logged each result under debug and dereferenced `result.Job`.

# How the Problems Are Solved

- Skip `nil` results and results with a `nil` `Job` before debug detail logging in the queue insert middleware.
- Add a regression test that exercises InsertManyFast-shaped nil results (and mixed nil/populated results) under debug and info log levels and asserts the middleware does not panic.

# Additional Changes

- Set the instance ID on the execution worker context via `authz.WithInstanceID` in `ContextWithExecuter` so workers have a non-empty instance for follow-up work.
- Make `Queue.Insert` / `InsertManyFastTx` nil-safe when the queue or River client is unset (defensive no-op).

# Additional Context

- Closes #12225
- A full API E2E with debug logging is not covered by the shared integration suite (`apps/api/test-integration-api.yaml` uses `Log.Level: info`); the package-level middleware test is the practical regression for the panic path.

### CODE REVIEW

File: internal/queue/queue.go [hunk 1]
```diff
@@ -124,6 +127,9 @@ func (q *Queue) Insert(ctx context.Context, args river.JobArgs, opts ...InsertOp
 //
 // Opts are applied to each job before sending them to river.
 func (q *Queue) InsertManyFastTx(ctx context.Context, tx *sql.Tx, args []river.JobArgs, opts ...InsertOpt) error {
+	if q == nil || q.client == nil {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (RmdanJr) [reply internal/queue/queue.go:134]: Fixed; Review again.

File: internal/queue/queue.go [hunk 2]
```diff
@@ -115,6 +115,9 @@ func WithQueueName(name string) InsertOpt {
 }
 
 func (q *Queue) Insert(ctx context.Context, args river.JobArgs, opts ...InsertOpt) error {
+	if q == nil || q.client == nil {
+		return nil
+	}
 	_, err := q.client.Insert(ctx, args, applyInsertOpts(opts))
 	return err
 }
```
[the remark this answers is not in the document]
Comment (RmdanJr) [reply internal/queue/queue.go:124]: Fixed; Review again.

File: internal/queue/queue.go [hunk 3]
```diff
@@ -115,6 +115,10 @@ func WithQueueName(name string) InsertOpt {
...
 
 func (q *Queue) Insert(ctx context.Context, args river.JobArgs, opts ...InsertOpt) error {
+	if q == nil || q.client == nil {
+		logging.Info(ctx, "skip inserting job because queue is not set")
```
[the remark this answers is not in the document]
Comment (RmdanJr) [reply internal/queue/queue.go:119]: In other similar logs, it is "not set". That's why I added it like this. I will not apply this change for consistency.

File: internal/queue/queue.go [hunk 4]
```diff
@@ -124,6 +128,10 @@ func (q *Queue) Insert(ctx context.Context, args river.JobArgs, opts ...InsertOp
 //
 // Opts are applied to each job before sending them to river.
 func (q *Queue) InsertManyFastTx(ctx context.Context, tx *sql.Tx, args []river.JobArgs, opts ...InsertOpt) error {
+	if q == nil || q.client == nil {
+		logging.Info(ctx, "skip inserting many jobs because queue is not set")
```
[the remark this answers is not in the document]
Comment (RmdanJr) [reply internal/queue/queue.go:132]: In other similar logs, it is "not set". That's why I added it like this. I will not apply this change for consistency.

**Comment by RmdanJr** [comment]
@peintnermax
Could you please check this?
